core/org.openjdk.jmc.common/src/main/java/org/openjdk/jmc/common/util/Environment.java

```diff
@@ -133,11 +132,7 @@ public static int getThisPID() {
 
 	private static int initPID() {
 		try {
-			String name = ManagementFactory.getRuntimeMXBean().getName();
-			if (name != null) {
-				String s = name.split("@")[0]; //$NON-NLS-1$
-				return Integer.parseInt(s);
-			}
+			return (int) ProcessHandle.current().pid();
 		} catch (Exception e) {
```

Comment:
I think the try/catch is not needed with ProcessHandle any more...

Reply:
Should I remove this still?

Reply:
Sure, or do you think it is needed?

Reply:
Well, theoretically the method can throw an exception. With security managers in place for example, but this is deprecated for removal in https://openjdk.org/jeps/411 so I'd go for removal.

Reply:
Yep. But there's no checked exception, so we shouldn't cater for something theoretical here. 😄